azure-workload-identity-webhook.yaml

```diff
@@ -0,0 +1,56 @@
+package:
+  name: azure-workload-identity-webhook
+  version: 1.5.0
+  epoch: 0
+  description: Mutating webhook for Azure Workload Identity
+  copyright:
+    - license: MIT
+
+environment:
+  environment:
+    CGO_ENABLED: "0"
+    GOFLAGS: "-trimpath"
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/Azure/azure-workload-identity
+      tag: v${{package.version}}
+      expected-commit: 350cb28ceab55d0696bef2077a1cc9dff368e982
+
+  - uses: go/build
+    with:
+      packages: ./cmd/webhook
+      output: manager
+      ldflags: |-
+        -X github.com/Azure/azure-workload-identity/pkg/version.BuildVersion=v${{package.version}}
+        -X github.com/Azure/azure-workload-identity/pkg/version.BuildTime=$(date -u +%Y-%m-%d-%H:%M)
+        -X github.com/Azure/azure-workload-identity/pkg/version.VCSVersion=$(git rev-parse --short HEAD)
+        -X github.com/Azure/azure-workload-identity/pkg/webhook.ProxyImageVersion=v${{package.version}}
+        -X github.com/Azure/azure-workload-identity/pkg/webhook.ProxyImageRegistry=mcr.microsoft.com/oss/azure/workload-identity
+
+  - uses: strip
+
+subpackages:
+  - name: ${{package.name}}-compat
+    description: Compatibility symlink for ${{package.name}}
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}"
+          ln -s /usr/bin/manager "${{targets.contextdir}}/manager"
+    test:
+      pipeline:
+        - runs: |
+            stat /manager
+
+update:
+  enabled: true
+  github:
+    identifier: Azure/azure-workload-identity
+    strip-prefix: v
+
+test:
+  pipeline:
+    - name: "Run manager and verify help output"
+      runs: |
+        manager --help
```

Comment:
can we expand on tests here, maybe by using `test/kwok/cluster` pipeline and then running the binary in a pseudo-kubernetes environment.